### PR TITLE
feature(logger): allow to print object

### PR DESCRIPTION
**Pull request by rluvaton**
BREAKING CHANGE: the logger now gets an optional object in its 2nd parameter

After this I'll make a PR to log the current context

### CODE REVIEW

File: src/code-templates/libraries/logger/index.ts
Comment (rluvaton) [inline src/code-templates/libraries/logger/index.ts:5]: Used JavaScript privates as they are really private unlike TypeScript private

File: src/code-templates/libraries/logger/pino.logger.ts
Comment (rluvaton) [inline src/code-templates/libraries/logger/pino.logger.ts:56]: Not just always doing
```
this.#logger.<method>(metadata, message);
```
as it won't print the message when the metadata is undefined...

----

Tiny dive into Pino code 😄 

This is happening cause Pino expects the first parameter to be either an object or a string:
And when passing `undefined` it passes to the `format` function `undefined`:
https://github.com/pinojs/pino/blob/dc4e8546c0a4bedb087cbcccd9a48b56100b0ad7/lib/tools.js#L56

and the format function just returns the first parameter if the type is not a string
https://github.com/pinojs/quick-format-unescaped/blob/20ebf64c2f2e182f97923a423d468757b9a24a63/index.js#L21-L23

File: src/code-templates/libraries/logger/test/index.test.ts
Comment (rluvaton) [inline src/code-templates/libraries/logger/test/index.test.ts:86]: Remove this after merging #226

File: src/code-templates/libraries/logger/index.ts
Comment (goldbergyoni) [inline src/code-templates/libraries/logger/index.ts:25]: Minor: Shall we maybe include getUnderlyingLogger function that will throw and stdout if logger instance does not exist? Theoretically, the user might call resetLogger + info

```getUnderlyingLogger().info(message, metadata);```

File: src/code-templates/libraries/logger/definition.ts
Comment (goldbergyoni) [inline src/code-templates/libraries/logger/definition.ts:4]: Why not ```metadata? : {}```?

I'm not suggesting rather asking, not sure what exactly Pino is expecting for

If Record, why not <unknown, unknown>?

cc @mikicho

File: src/code-templates/libraries/logger/index.ts
Comment (goldbergyoni) [reply src/code-templates/libraries/logger/index.ts:5]: I don't have a strong opinion here, just remember that sometimes 

readability > correctness

**Review by goldbergyoni — COMMENTED** [review]
@rluvaton Great fix, NOW our logger conforms to important best practices

File: src/code-templates/libraries/logger/definition.ts
Comment (mikicho) [reply src/code-templates/libraries/logger/definition.ts:4]: metadata: Object would make sense here, but there are subtle differences between the two so need to check .. but I prefer metadata: Object

Comment (goldbergyoni) [reply src/code-templates/libraries/logger/definition.ts:4]: Big O' Object means it must inherit from Object.proptotype which is not the case here

Record feels to me like an overkill, we want to educate for simplicity

```metadata: object``` feels simple, sufficient, and aligned with Pino's expectation

File: src/code-templates/libraries/logger/pino.logger.ts
Comment (goldbergyoni) [reply src/code-templates/libraries/logger/pino.logger.ts]: @rluvaton You're not leaving blind spots, gotta love this

File: src/code-templates/libraries/logger/index.ts
Comment (rluvaton) [reply src/code-templates/libraries/logger/index.ts:25]: we call configure logger in line 24

File: src/code-templates/libraries/logger/definition.ts
Comment (rluvaton) [reply src/code-templates/libraries/logger/definition.ts:4]: I used Record as I remember having but here I don't get it so I'll use object :)
```
Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys.eslint@typescript-eslint/ban-types
```

Comment (mikicho) [reply src/code-templates/libraries/logger/definition.ts:4]: [Not sure if the differences are relevant here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBBMF4YGUCeBbARiANgCgHIBDAgSgFgAoUSWAByNRxCIBNEYBvOALhgG0AjAF0AvlQBmAVzDAoAS3Ax5YCSABKAU1AAnVnnSaIEIgHNNfaDpWmANDENQ2RJwH4+W3awA8Vm-aIwVAA+Ui4YGghcTQA6ZlMDTSdWFyIw8UoVNU8QPUIAC01UAnsGJhZWCkpJGTlFMGVVEAB5TAArbShE4zMLOChrMDsHJOc3PlaOuTDuSOi4kATHMbSYDKyW9s6CopKYMuY2KprZBSUNkC25bpNzSwH-EeTU9xhLqagZiPAonFj4xLPJzpKgXK5dAiFYqlRiHSpAA). I think we should go with simplicity/readability. if issue will raise in the future, we'll tackle it than

Comment (rluvaton) [reply src/code-templates/libraries/logger/definition.ts]: agree and updated

File: src/code-templates/libraries/logger/index.ts
Comment (goldbergyoni) [reply src/code-templates/libraries/logger/index.ts:25]: What guarantee you have the it will get called? There should be no trust between order-service and logger

Also, if you put this function, no need to to put "!" in every call to logger - this function will return a guaranteed instance of logger

Your call, don't pause because of this

**Review by goldbergyoni — APPROVED** [review]
Awesome work by Raz Luvaton, left 1 comment - don't pause, make a call and merge

File: src/code-templates/libraries/logger/index.ts
Comment (rluvaton) [reply src/code-templates/libraries/logger/index.ts:25]: I need to put `!` anyway as the property is nullable

I'm sure the configure is getting called as I call this in each logging method (i.e inside info, warn and etc...)

**Pull request merged**
